### ISSUE TITLE
removing remotes

### ISSUE THREAD

**Issue by reider-dev** [zed-industries/zed#15910]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

```
command pallet -> projects: open remote
```
A nice popup titled "Remote Projects (alpha)" with a list of servers. When clicking the little trash icon to remove the server an "are you sure" confirmation appears. After clicking "Delete" a "Failed to delete dev server: error upgrading view" feedback is given.

I tried to remove zed and reinstall, but the remote was somehow still listed (I am not sure where it caches the remotes info, since I seemed to have failed when deleting all folders with zed in the name).

I would have added screenshots, but it seems there is some challenge with screenshotting zed on my platform (gnome, fedora, wayland, ?).

Any help is appreciated, and I can certainly try things out and provide timely feedback.

### Environment

Zed: v0.146.5 (Zed)
OS: Linux Wayland fedora 40
Memory: 62.4 GiB
Architecture: x86_64
GPU: Intel(R) Graphics (ADL GT2) || Intel open-source Mesa driver || Mesa 24.1.5

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

2024-08-07T16:20:32.482313128+02:00 [INFO] rebuilt extension index in 81.392768ms
2024-08-07T16:20:32.482603338+02:00 [INFO] extensions updated. loading 0, reloading 1, unloading 0
2024-08-07T16:20:32.789804253+02:00 [WARN] request completed with error: failed to connect to the server
2024-08-07T16:21:08.929238334+02:00 [ERROR] error upgrading view
2024-08-07T16:21:35.488562875+02:00 [ERROR] error upgrading view
</pre></details>

**Comment by JosephTLyons** [comment]
cc @ConradIrwin

**Comment by notpeter** [comment]
I wasn't able to reproduce this on my Ubuntu Linux 22 machine.
I can add/remove remotes without issue.

To delete Zed data and start fresh you can delete `~/.local/share/zed`.  I believe the remote servers are stored under `db/` so that may be all you need to remove.  Note, this directory also contains `conversions` and `prompts` for the AI assistant which are only stored here so preserve that data first if desired.  Make sure zed is closed when you delete this folder.

**Comment by weiji14** [comment]
I'm encountering the same "Failed to delete dev server project: error upgrading view. Please try again" pop-up when trying to delete remote projects. Using Debian Linux 13 with Zed 0.148.0.

![image](https://github.com/user-attachments/assets/d69801d5-dbb1-4805-a39e-95445717025d)

**Comment by reiderbelimo** [comment]
> I wasn't able to reproduce this on my Ubuntu Linux 22 machine with Zed v0.148.x I can add/remove remotes without issue.
> 
> To delete Zed data and start fresh you can delete `~/.local/share/zed`. I believe the remote servers are stored under `db/` so that may be all you need to remove. Note, this directory also contains `conversions` and `prompts` for the AI assistant which are only stored here so preserve that data first if desired. Make sure zed is closed when you delete this folder.

Thanks for the db location hint..  It seems to do the trick, but I also "signed out" first, so unsure if it had an influence. Since I don't have any important "conversations" or "prompts", it was a very low risk thing to try :)

**Comment by kauanfontanela** [comment]
> To delete Zed data and start fresh you can delete `~/.local/share/zed`.

As soon as you log in the remote servers are listed again. It kinda concerns me that they are stored on Zed servers and I am not able to remove them since I also get the following error on doing so:
> Failed to delete dev server project: error upgrading view. Please try again

**Comment by notpeter** [comment]
We don't store your remote ssh connections on our servers.

I'm pretty sure that's stored under `~/.local/share/zed/db`

Did you perhaps delete it while zed was running and then close (triggering a save) and then relaunch?

**Comment by kauanfontanela** [comment]
@notpeter I made sure zed-editor was not running and deleted the whole `~./local/share/zed` directory. The remote projects list did go away, but as soon as I logged in they came back.

@ConradIrwin isn't there any way I could request for my connection strings to be deleted from zed's server? Since deleting from the editor seems not to be working:

![image](https://github.com/user-attachments/assets/57f46d92-f921-4fde-901a-e2a1d06b41b5)
